rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -133,96 +112,43 @@ public static boolean isNotPartOfRunningBuild(FlowExecution execution) {
         return (execution != null && execution.isComplete());
     }
 
-    public static long getNodeExecDuration(FlowNode node) {
-        long startTime = TimingAction.getStartTime(node);
-        if (startTime == 0L) {
-            // The node is running and the time has not been marked on it yet.  Return 0 as the duration for now.
-            return 0L;
-        }
-
-        // All we need to calculate exec duration is the last child nodes
-        FlowNode lastChild = getLastChildNode(node);
-        if (lastChild != null) {
-            long endTime = TimingAction.getStartTime(lastChild);
-            return (endTime - startTime);
-        } else {
-            return 0L;
-        }
-    }
-
-    public static FlowNode getLastChildNode(FlowNode node) {
-        List<FlowNode> allNodes = getIdSortedExecutionNodeList(node.getExecution());
-        for(int i=allNodes.size()-1; i>=0; i--) {
-            FlowNode f = allNodes.get(i);
-            if (f.getParents().contains(node)) {
-                return f;
-            }
+    /** Find a node following this one */
+    @CheckForNull
+    public static FlowNode getNodeAfter(@Nonnull final FlowNode node) {
+        if (node.isRunning()) {
+            return null;
         }
-        return null;
-    }
-
-    @edu.umd.cs.findbugs.annotations.SuppressWarnings(value="NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
-        justification = "Precondition of block ensures that null cannot be returned (FlowExecution checked)")
-    public static ExecDuration getStageExecDuration(FlowNode stageStartNode) {
-        FlowExecution execution = stageStartNode.getExecution();
-        if (execution != null && StageNodeExt.isStageNode(stageStartNode)) {
-            List<FlowNode> allNodesSorted = getIdSortedExecutionNodeList(execution);
-            int stageStartNodeIndex = findStageStartNodeIndex(allNodesSorted, stageStartNode);
-            FlowNode firstExecutedNode;
-
-            // Locate the first node in the stage that was actually executed.  This can
-            // vary e.g. when there's was a checkpoint restart, the first executed node
-            // in the stage could be in middle of the stage Pipeline definition.
-            firstExecutedNode = getStageExecStartNode(allNodesSorted, stageStartNodeIndex);
 
-            if (firstExecutedNode != null) {
-                long startTime = TimingAction.getStartTime(firstExecutedNode);
-                long endTime;
+        int iota = Integer.parseInt(node.getId());
```

Comment:
I will treat this as a :bug: since there is no guarantee that IDs are in fact numeric, though in the current implementation they happen to be. (In fact it would be a nice idea to convert them to a dotted-tree notation.)